src/cmd/build.js

```diff
@@ -95,11 +95,23 @@ export function safeFileName(name: string): string {
 export class FileFilter {
   filesToIgnore: Array<string>;
 
-  constructor({filesToIgnore}: Object = {}) {
+  constructor(artifactsDir,{filesToIgnore}: Object = {}) {
+    var eliminateArtifactDir =  new String();
+    var buf = artifactsDir;
+    if(typeof buf !== 'undefined' && buf.indexOf('web-ext-artifacts') != -1){
+        eliminateArtifactDir = buf.slice(buf.indexOf('web-ext-artifacts'));
+    }
+    else if(typeof buf !== 'undefined' && buf.slice(-1) === '/'){
+	eliminateArtifactDir = path.join(buf.slice(0,-1));
+    }
+    else if(typeof buf !== 'undefined'){
+	eliminateArtifactDir =  path.join(buf);
+    }
     this.filesToIgnore = filesToIgnore || [
       '**/*.xpi',
       '**/*.zip',
       '**/.*', // any hidden file
+      '**/*'+eliminateArtifactDir,
```

Comment:
I think this would be easier if you made a separate array, like, `this.filePathsToIgnore` and then checked the exact paths in a loop separate from the minimatch loop. Example:

````javascript
wantFile(path: string): boolean {
  for (const pathToIgnore of this.filePathsToIgnore) {
    if (path === pathToIgnore) {
      return false;
    }
  }
  for (const test of this.filesToIgnore) {
    if (minimatch(path, test)) {
      // no change to this...
    }
  }
}
````